.changeset/lovely-walls-guess.md

```diff
@@ -0,0 +1,90 @@
+---
+'@backstage/create-app': patch
+---
+
+Added `SidebarSubmenu` for all Kinds making it easier to discover them in the Catalog.
+
+To add this to an existing Backstage instance you will need to make the following to your `packages/app/src/components/Root/Root.tsx`:
+
+1. First add the following imports:
+
+   ```ts
+   import ApiIcon from '@material-ui/icons/Extension';
+   import ComponentIcon from '@material-ui/icons/Memory';
+   import DomainIcon from '@material-ui/icons/Apartment';
+   import ResourceIcon from '@material-ui/icons/Work';
+   import SystemIcon from '@material-ui/icons/Category';
+   import UserIcon from '@material-ui/icons/Person';
+   ```
+
+2. Then update the `@backstage/core-components` import like this:
+
+   ```diff
+     import {
+       Sidebar,
+       sidebarConfig,
+       SidebarDivider,
+       SidebarGroup,
+       SidebarItem,
+       SidebarPage,
+       SidebarScrollWrapper,
+       SidebarSpace,
+       useSidebarOpenState,
+       Link,
+   +   GroupIcon,
+   +   SidebarSubmenu,
+   +   SidebarSubmenuItem,
+     } from '@backstage/core-components';
+   ```
+
+3. Finally replace `<SidebarItem icon={HomeIcon} to="catalog" text="Home" />` with this:
+
+   ```ts
+   <SidebarItem icon={HomeIcon} to="catalog" text="Home">
+     <SidebarSubmenu title="Catalog">
+       <SidebarSubmenuItem
+         title="Domains"
+         to="catalog?filters%5Bkind%5D=domain"
```

Comment:
QQ: does `filters[kind]=domain` work here too? just reads a little nicer if it doesn't break the link

Reply:
I believe so, I just grabbed how they were in the URL when I set these up and seeing as they worked on our instance just ran with that. I'll clean these up later this week.

Reply:
OK, confirmed that does work and it's much nicer to read so I've made the change for this.